File: Data Structures & Algorithms/reverse-string/submission-3.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        # Method1 two pointer
        # i=0
        # j=len(s)-1
        # while(i<j):
        #     s[i],s[j]=s[j],s[i]
        #     # temp=s[i]
        #     # s[i]=s[j]
        #     # s[j]=temp
        #     i+=1
        #     j-=1
        # method2 stack
        stack=[]
        for c in s:
            stack.append(c)
        i=0
        while stack:
            s[i]=stack.pop()
            i+=1

File: Data Structures & Algorithms/validate-parentheses/submission-3.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for(char &str: s){
            if(str=='{' || str=='('|| str=='['){
                stack.push(str);
            }
            else{
                if(str=='}' && !stack.empty()){
                    if(stack.top()=='{')
                    {stack.pop();}
                    else{
                    return false;
                }
                }
                 else  if(str==')' && !stack.empty()){
                    if(stack.top()=='(')
                    {stack.pop();}
                    else{
                    return false;
                }
                }
                
                 else  if(str==']' && !stack.empty()){
                    if(stack.top()=='[')
                    {stack.pop();}
                    else{
                    return false;
                }
                }
                else{
                    return false;
                }
                
            }
        }
        
            return stack.empty();
        
    }
};
